src/main/java/com/github/Anon8281/universalScheduler/foliaScheduler/FoliaScheduledTask.java

```diff
@@ -4,11 +4,20 @@
 import io.papermc.paper.threadedregions.scheduler.ScheduledTask;
 import org.bukkit.plugin.Plugin;
 
+import java.util.HashMap;
+import java.util.Random;
+
 public class FoliaScheduledTask implements MyScheduledTask {
     private final ScheduledTask task;
+    static final HashMap<Integer, ScheduledTask> tasks = new HashMap<>();
+    private int taskId;
 
     public FoliaScheduledTask(final ScheduledTask task) {
         this.task = task;
+        do {
+            taskId = new Random().nextInt();
+        } while (tasks.containsKey(taskId));
+        tasks.put(taskId, task);
```

Comment:
Pretty sure that there is a possible memory leak. As you adding the task to the Map, you should remove it from this map as soon as the task will be done. 
Otherwise, in case when your server is up enough time, you will get OOM error, because a lot of task created and noone task removed.